pkg/upgrade/upgrade.go

```diff
@@ -0,0 +1,289 @@
+package upgrade
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/weaveworks/pctl/pkg/bootstrap"
+	"github.com/weaveworks/pctl/pkg/catalog"
+	"github.com/weaveworks/pctl/pkg/git"
+	"github.com/weaveworks/pctl/pkg/install"
+	"github.com/weaveworks/pctl/pkg/runner"
+	go_git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport"
+	gitssh "gopkg.in/src-d/go-git.v4/plumbing/transport/ssh"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+type UpgradeValues struct {
+	RepoOrgAndName string
+	Remote         string
+	HeadBranch     string
+	BaseBranch     string
+	CommitMessage  string
+	Name           string
+	Namespace      string
+	ProfileBranch  string
+	ConfigMap      string
+	Out            string
+	ProfileRepoURL string
+	ProfilePath    string
+	GitRepository  string
+	Version        string
+}
+
+func Upgrade(upgradeValues UpgradeValues, w io.Writer) error {
+	config := config.GetConfigOrDie()
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(w, "Checking if entitlement exists...\n")
+
+	entitlement, err := getEntitlement(clientset)
+	if err != nil {
+		return err
+	}
+
+	upgradeValues.Name = "weave-gitops-enterprise"
+	upgradeValues.Namespace = "wego-system"
```

Comment:
This may be a misunderstanding on my part with how we are handling things in enterprise but in install we can pass in a namespace and do the install on that namespace.
```
func installRunCmd(cmd *cobra.Command, args []string) error {
	namespace, _ := cmd.Parent().Flags().GetString("namespace")
```
It looks to me like if someone did install in their own namespace it would screw up the upgrade.

Reply:
I agree! We can make the namespace configurable and give it a default value 👍

Reply:
That totally makes sense to me! Thank you 🙏